### PR TITLE
feat: rewrite indexing to Go

### LINKED ISSUE #260: Rewrite MeiliSearch indexing in Go
### Is your feature request related to a problem? Please describe.

The MeiliSearch indexing step is by far the slowest now that we moved the API fetching to Go 🙏. Let's repeat the success!

### Describe the solution you'd like

I'd like an endpoint in the Internal service that triggers the MeiliSearch indexing. It should probably just take the data from Postgres on its own instead of getting all of the data passed over HTTP, but whoever makes it can decide

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
# Description

Please include a summary of the change and which issue is fixed.

Fixes #260 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: internal/types.go [hunk 1]
```diff
@@ -66,26 +112,54 @@ type TV struct {
...
+		Genres:                     genres,
+		PosterPath:                 tv.PosterPath,
+		Popularity:                 tv.Popularity,
+		SupportedProviderCountries: getSupportedProviderCountries(tv.Providers),
+		Providers:                  tv.Providers,
 	}
 }
 
 type Provider struct {
-	Results map[string]struct {
```
Comment (AndreasPB) [inline internal/types.go:88]: Should probably still be named Results

**Comment by AndreasPB** [comment]
It still breaks :(

File: internal/types.go [hunk 2]
```diff
@@ -105,6 +113,32 @@ type MediaIds struct {
...
 }
 
-type Env struct {
-	db *gorm.DB
+func getMediaType(id string) string {
+	if id[0:1] == "m" {
+		return "movie"
+	}
+	return "tv"
+}
```
Comment (AndreasPB) [inline internal/types.go:121]: Should probably handle "t" as well, for when we get people. Throw an easy to understand error if it's not m or t 🗡️

**Review by AndreasPB — APPROVED** [review]

File: internal/types.go (hunk 2 above)
Comment (nullxDEADBEEF) [reply internal/types.go]: Done

**Pull request merged**